test/test_query.py

```diff
@@ -113,6 +113,30 @@ def test_where(self):
             Query(t).where(t.numField > 4).render(),
         )
 
+    def test_where_multiple_predicates(self):
+        self.assertEqual(
+            'test_table | where boolField and (numField > numField2) and (stringField contains "hello")',
+            Query(t).where(t.boolField, t.numField > t.numField2, t.stringField.contains('hello')).render(),
+        )
+
+    def test_where_no_predicates(self):
+        self.assertEqual(
+            'test_table | project numField',
+            Query(t).where().project(t.numField).render(),
+        )
+
+    def test_where_trivial_predicate(self):
+        self.assertEqual(
+            'test_table | where boolField | project numField',
+            Query(t).where(t.boolField, True).project(t.numField).render(),
+        )
+
+    def test_where_only_trivial_predicate(self):
```

Comment:
add test for not(boolField), it is broken now (thought of opening an issue, but this seems close enough

Reply:
The below assert works. So what is broken exacrly?

    self.assertEqual(
        "test_table | where not(boolField)",
        Query(t).where(~t.boolField).render(),
    )

Reply:
I tried using "not" and "!", and it failed, but it could be just syntax we won't allow.
In that case, I think we need to make it more clear that this is the way we negate (add to docs?).
Anyway, since the functionality does exist, it's not that severe.

Reply:
The `!` operator does not exist in Python.
Regarding `not` - unfortunately it cannot be overridden in Python (and neither can `and` and `or`, that's why we use the bitwise operators `&`, `|` and `~`).
We can add a comment about it in the docs, but no one will read it. To make things worse, when you try `where(not t.boolField)` there is no error message, because Python evaluates `not t.boolField` to be `False`, and we get the query `| where false`. This could lead to serious bugs, but I don't know how to help people avoid it.

Reply:
One possibility is to forbid calling `where` with `False` as the argument - because why would anyone want to do that?
However, there are use cases for this, if one wants to calculate the predicates in runtime:
`where(some_python_code_that_might_return_false)`

Reply:
I added some documentation, and a `not_of` method

Reply:
looks good

Reply:
Update: I managed to find a way to prevent accidental use of logical operators. See here:
https://github.com/Azure/pykusto/pull/97